inc/cl-helpers.php

```diff
@@ -121,3 +121,124 @@ function uri_cl_get_vimeo_thumbnail( $src ) {
+	$year_string = 1 == $years ? 'year' : 'years';
+	$months = round( $days / 30 );
+	$month_string = 1 == $months ? 'month' : 'months';
+
+	if ( 0 == $days ) {
+		// if we want to get really granular, this will do it.
+		// caching more or less defeats the purpose of getting so granular
+		// if ( $hours > 1 ) {
+		// $message = sprintf('<span class="time-left-number">%d</span> %s until <span class="time-left-event">%s</span>.', $hours, $hour_string, $attributes['event'] );
+		// } elseif ( $minutes > 0 ) {
+		// $message = sprintf('<span class="time-left-number">%d</span> %s until <span class="time-left-event">%s</span>.', $minutes, $minute_string, $attributes['event'] );
+		// } else {
+		// $message = sprintf('<span class="time-left-number">%d</span> %s until <span class="time-left-event">%s</span>.', $seconds, $second_string, $attributes['event'] );
+		// }
+		$capitalized_event = ucfirst( $attributes['event'] );
+		$message = sprintf( '<span class="time-left-event">%s</span> %s.', $capitalized_event, $attributes['is_today'] );
+	} elseif ( $days > 59 ) {
+		if ( $years > 0 ) {
+			$message = sprintf( '<span class="time-left"><span class="time-left-number">%d</span> <span class="time-left-unit">%s</span></span> %s <span class="time-left-event">%s</span>.', $years, $year_string, $attributes['until'], $attributes['event'] );
+		} else {
+			$message = sprintf( '<span class="time-left"><span class="time-left-number">%d</span> <span class="time-left-unit">%s</span></span> %s <span class="time-left-event">%s</span>.', $months, $month_string, $attributes['until'], $attributes['event'] );
+		}
+	} else {
+		$message = sprintf( '<span class="time-left"><span class="time-left-number">%d</span> <span class="time-left-unit">%s</span></span> %s <span class="time-left-event">%s</span>.', $days, $day_string, $attributes['until'], $attributes['event'] );
+	}
+
+	if ( $days < 0 ) {
+		if ( false !== $attributes['show_expired'] ) {
+			if ( $days < -30 ) {
+				if ( $years < 0 ) {
+					$message = sprintf( '<span class="time-left-event">%s</span> %s <span class="time-left"><span class="time-left-number">%d</span> <span class="time-left-unit">%s</span></span> ago.', $attributes['event'], $attributes['passed'], abs( $years ), $year_string );
+				} else {
+					$message = sprintf( '<span class="time-left-event">%s</span> %s <span class="time-left"><span class="time-left-number">%d</span> <span class="time-left-unit">%s</span></span> ago.', $attributes['event'], $attributes['passed'], abs( $months ), $month_string );
+				}
+			} elseif ( $days < -1 ) {
+				$message = sprintf( '<span class="time-left-event">%s</span> %s <span class="time-left"><span class="time-left-number">%d</span> <span class="time-left-unit">%s</span></span> ago.', $attributes['event'], $attributes['passed'], abs( $days ), $day_string );
+			} else {
+				$message = sprintf( '<span class="time-left-event">%s</span> has %s.', $attributes['event'], $attributes['passed'] );
+			}
+		} else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method uri_cl_shortcode_get_template uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://app.codacy.com/app/uriweb/uri-component-library/pullRequest?prid=4536390)